services/relayer/internal/msgobserver/eth/observer/observer_impl_v1.go

```diff
@@ -123,7 +135,14 @@ func (o *ObserverImplV1) StartObserve(chainID *big.Int, chainAP string, contract
 	if err != nil {
 		return err
 	}
-	go o.routine(chainID, chainAP, contractAddr)
+	if contractType == "SFC" {
+		go o.routineSFC(chainID, chainAP, contractAddr)
+	} else if contractType == "GPACT" {
```

Comment:
nit: should these have similar case variation checks to lines 97 and 99?

Reply:
Yea. Maybe better to use a different way. Will do this in next PR